cloud/edgecontroller/pkg/devicecontroller/apis/devices/v1alpha1/device_instance_types.go

```diff
@@ -0,0 +1,160 @@
+	Protocol ProtocolConfig `json:"protocol,omitempty"`
+	// NodeSelector indicates the binding preferences between devices and nodes.
+	// Refer to k8s.io/kubernetes/pkg/apis/core NodeSelector for more details
+	// +optional
+	NodeSelector *core.NodeSelector `json:"nodeSelector,omitempty"`
+}
+
+// Only one of its members may be specified.
+type ProtocolConfig struct {
+	// Protocol configuration for opc-ua
+	// +optional
+	OpcUA *ProtocolConfigOpcUA `json:"opcua,omitempty"`
+	// Protocol configuration for modbus
+	// +optional
+	Modbus *ProtocolConfigModbus `json:"modbus,omitempty"`
+}
+
+type ProtocolConfigOpcUA struct {
+	// Required: The URL for opc server endpoint.
+	Url string `json:"url,omitempty"`
+	// Username for access opc server.
+	// +optional
+	UserName string `json:"userName,omitempty"`
+	// Password for access opc server.
+	// +optional
+	Password string `json:"password,omitempty"`
+	// Defaults to "none".
+	// +optional
+	SecurityPolicy string `json:"securityPolicy,omitempty"`
+	// Defaults to "none".
+	// +optional
+	SecurityMode string `json:"securityMode,omitempty"`
+	// Certificate for access opc server.
+	// +optional
+	Certificate string `json:"certificate,omitempty"`
+	// PrivateKey for access opc server.
+	// +optional
+	PrivateKey string `json:"privateKey,omitempty"`
+	// Timeout seconds for the opc server connection.???
+	// +optional
+	Timeout int64 `json:"timeout,omitempty"`
+}
+
+// Only one of its members may be specified.
+type ProtocolConfigModbus struct {
+	// +optional
+	RTU *ProtocolConfigModbusRTU `json:"rtu,omitempty"`
+	// +optional
+	TCP *ProtocolConfigModbusTCP `json:"tcp,omitempty"`
+}
+
+type ProtocolConfigModbusTCP struct {
+	// Required.
+	IP string `json:"ip,omitempty"`
+	// Required.
+	Port int64 `json:"port,omitempty"`
+	// Required.
+	SlaveID string `json:"slaveID,omitempty"`
+}
+
+type ProtocolConfigModbusRTU struct {
+
+	// Required.
+	SerialPort string `json:"serialPort,omitempty"`
+	// Required. BaudRate 115200|57600|38400|19200|9600|4800|2400|1800|1200|600|300|200|150|134|110|75|50
+	BaudRate int64 `json:"baudRate,omitempty"`
+	// Required. Valid values are 8, 7, 6, 5.
+	DataBits int64 `json:"dataBits,omitempty"`
+	// Required. Valid options are "none", "even", "odd". Defaults to "none".
+	Parity string `json:"parity,omitempty"`
+	// Required. Bit that stops 1|2
+	StopBits int64 `json:"stopBits,omitempty"`
+	// Required. 0-255
+	SlaveID int64 `json:"slaveID,omitempty"`
+}
+
+// DeviceStatus reports the device state and the desired/reported values of twin attributes.
+type DeviceStatus struct {
+	// A list of device twins containing desired/reported desired/reported values of twin properties..
+	// Optional: A passive device won't have twin properties and this list could be empty.
+	// +optional
+	Twins []Twin `json:"twins,omitempty"`
+}
+
+// A Twin provides a logical representation of control properties (writable properties in the
```

Comment:
Remove `A` to get golint check passed in the future we add static checks in CI.

Reply:
updated